client/src/components/inventory/RequestButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import {Button, Icon} from "semantic-ui-react";
+import {useMutation, useQuery} from "@apollo/react-hooks";
+import {CREATE_REQUEST} from "../util/graphql/Mutations";
+import {RequestedItem} from "../../types/Hardware";
+import {GET_USER} from "../util/graphql/Queries";
+
+interface RequestButtonProps {
+    request: RequestedItem,
+}
+
+function RequestButton({request}: RequestButtonProps) {
+    const {subscribeToMore, ...query} = useQuery(GET_USER);
+    const [createRequest, {data, loading, error}] = useMutation(CREATE_REQUEST);
+
+    return (
+
+        <Button primary
+                icon
+                disabled={loading}
+                loading={loading}
+                onClick={event => createRequest({
+                    variables: {
+                        newRequest: {
+                            user_id: query.data.user.uuid,
+                            request_item_id: request.id,
```

Comment:
naming the prop `request` here doesn't make sense semantically.  Can you make it clear that this is the ID of the item being requested?